[Merge] 优化 Redis 配置与权限管理逻辑

FLASHLACK1314 opened the pull request:
## PR 标题

[Merge] 优化 Redis 配置与权限管理逻辑

---

## 变更内容

- **新增/修改/删除**：
    - 新增 Redis 配置类 `RedisConfig`，用于初始化 Redis 连接。
    - 修改权限相关注解，调整了 `RequestPermission` 和 `RequestRole` 注解的实现方式。
    - 删除了 `AuthController.java` 和 `BuildingController.java` 中的部分方法，优化了权限控制和接口结构。
    - 修改了与 Redis 相关的缓存操作逻辑，改为使用 `Jedis` 替代 `Redisson` 进行缓存操作。
- **涉及模块**：
    - 后端（包括 Redis 配置、权限管理、用户管理）
    - 数据库（缓存操作、用户信息）
    - 配置（Redis 配置、权限控制）
- **关键实现**：
    - 更新了 Redis 连接配置，采用了 Hutool 的 `RedisDS` 进行连接管理。
    - 优化了 AOP 切面中的权限检查逻辑，增强了权限校验和角色验证。
    - 移除了部分已废弃的 Redis 操作方式（如 `Redisson`）并转为使用 `Jedis` 进行 Redis 连接和操作。

---

## 测试内容

- 描述已进行的测试，测试覆盖哪些场景。
    - 单元测试：
        - 测试了 Redis 连接是否正常，以及 `Jedis` 的缓存读取和写入。
        - 测试了修改后的权限注解和 AOP 切面是否正常工作。
    - 集成测试：
        - 测试了与数据库交互的功能，确认了 Redis 与数据库的数据同步是否正常。
    - 手动测试：
        - 手动检查了修改后的 API，确保接口的权限控制和 Redis 配置正常。
- 说明是否有新增或修改的测试用例，并提供测试步骤和结果：
    - 增加了 Redis 配置和权限验证的单元测试，确保无论是数据库访问还是权限校验都能正常工作。

---

## 相关问题

- 无

---

## 备注

- 为确保 Redis 连接稳定性，新增了 `RedisConfig` 类来管理连接池。
- 优化了权限验证逻辑，权限校验与角色校验分开处理，增加了异常处理机制。

---

## 截止日期

- 2025年3月1日

---

## 请在合并前确保

- [ ] 已完成代码审查
- [x] 通过所有测试
- [ ] 已更新文档（如有）
- [ ] 代码风格符合项目要求
- [ ] 已解决所有合并冲突

Commit:
feat(user): 添加获取用户信息功能

- 在 UserController 中新增 getUserInfo 接口
- 在 UserLogic 和 UserService 中实现 getUserInfo 方法- 该功能允许通过用户 UUID 获取用户详细信息，包括角色和相关实体信息

Commit:
feat(user): 添加用户功能
TODO:存在无法复现的bug，在缓存特别多的情况下出现使用UserDAO中的getUserBy的各种方法无法正常获取值的情况
- 新增 UserAddVO 类用于用户添加视图对象
- 在 UserController 中添加 addUser 方法处理用户添加请求
- 在 UserLogic 和 UserService 中实现用户添加的业务逻辑
- 优化 ProjectUtil 类，增加随机密码生成功能

Commit:
refactor(user): 优化用户添加功能

- 将 UserAddVO 中的 roleName 字段改为 roleUuid，用于获取角色信息
- 修改 addUser 方法，通过 roleUuid 获取角色

Commit:
feat(user): 添加用户删除功能并优化相关逻辑

- 在 UserController 中添加 deleteUser 方法，实现用户删除接口
- 在 UserLogic 中实现 deleteUser 方法的业务逻辑
- 在 UserDAO 中添加 deleteUser 方法，实现用户数据的物理删除
- 优化了 TokenDAO，增加 clearUserToken 方法清除用户相关缓存
- 重构了部分代码，提高了可维护性和可扩展性

Commit:
fix(user): 优化用户删除逻辑

- 在 UserDAO 中添加事务处理，确保用户数据和相关缓存同时删除
- 在 UserLogic 中添加 @Transactional 注解，确保业务逻辑原子性
- 异常处理使用自定义 ServerInternalErrorException，提高错误信息的准确性

Commit:
fix(user): 优化用户删除逻辑

- 在 UserDAO 中添加事务处理，确保用户数据和相关缓存同时删除
- 在 UserLogic 中添加 @Transactional 注解，确保业务逻辑原子性
- 异常处理使用自定义 ServerInternalErrorException，提高错误信息的准确性

Commit:
refactor(scheduling): 优化用户删除逻辑

- 在 StudentDAO 中新增 deleteStudent 方法，用于删除学生信息并清除相关缓存
- 更新 UserDAO 中的 deleteUser 方法，添加异常处理并优化缓存清理逻辑
- 修改 UserLogic 中的学生删除逻辑，使用新的 deleteStudent 方法
- 优化代码结构，提高可读性和可维护性

Commit:
feat(dao): 添加删除学生和教师的方法

- 在 StudentDAO 中添加 deleteStudent 方法，用于删除指定的学生信息
- 在 TeacherDAO 中添加 deleteTeacher 方法，用于删除指定的教师信息
- 修改 UserLogic 中的删除逻辑，使用新的删除方法

Commit:
feat(scheduling): 新增部门相关实体、Mapper和DAO

- 新增 DepartmentDO 部门信息实体类
- 新增 DepartmentMapper 部门 Mapper 接口
- 新增 DepartmentDAO部门 DAO 类
- 更新 MajorDO 实体类，增加专业所属学院等字段

Commit:
refactor(scheduling): 重构 MajorDAO 和 DepartmentDAO 类

- 在 DepartmentDAO 和 MajorDAO 类中添加 @Repository 注解
- 修改 MajorDO 类中的字段名，将 majorDepartmentUuid 改为 departmentUuid

Commit:
test(scheduling): 添加学生信息删除功能的单元测试

- 新增 StudentTest 类，实现学生信息删除的单元测试
- 测试包括保存学生信息、删除学生信息，并验证删除效果
- 使用 Redis进行数据缓存，并在测试中验证缓存数据的一致性
- 依赖 DepartmentDAO 和 MajorDAO 获取部门和专业信息

Commit:
test(scheduling): 优化学生信息相关测试代码

- 在保存学生信息前增加查询，避免重复插入
- 删除学生信息后，增加对 Redis 缓存的清理- 优化代码格式，统一使用字符串拼接方式

Commit:
feat(user): 添加用户编辑功能并优化用户相关操作

- 新增用户编辑功能，支持修改用户名、密码、邮箱、手机号等信息- 优化用户注册、登录、获取信息等接口的实现
- 增加了对用户角色和权限的处理
- 改进了用户数据的验证和错误处理机制

Commit:
Merge remote-tracking branch 'origin/develop' into feature/admin-manage-user

# Conflicts:
#	src/main/java/com/frontleaves/scheduling/constants/StringConstant.java
#	src/main/java/com/frontleaves/scheduling/daos/TokenDAO.java
#	src/main/java/com/frontleaves/scheduling/models/entity/MajorDO.java

Commit:
feat(user): 添加用户列表查询功能

- 在 UserController 中新增 getUserList 接口，用于获取用户列表
- 在 UserLogic 和 UserService 中添加 getUserList 方法，用于处理用户列表查询逻辑
- 使用 PageDTO<UserInfoDTO> 作为返回类型，支持分页查询

Commit:
feat(user): 添加用户列表查询功能

- 在 UserDAO 中新增 getUserList 方法，实现分页查询用户数据
- 在 UserLogic 中实现 getUserList 方法，处理用户列表查询逻辑
- 在 UserController 中添加 getUserList 接口，提供用户列表查询服务
- 在 TeacherDAO 和 StudentDAO 中分别新增根据用户 UUID 获取教师和学生信息的方法
- 优化了用户角色判断和相关信息获取的逻辑

Commit:
Merge branch 'develop' into feature/admin-manage-user

# Conflicts:
#	src/main/java/com/frontleaves/scheduling/controllers/UserController.java
#	src/main/java/com/frontleaves/scheduling/daos/TeacherDAO.java
#	src/main/java/com/frontleaves/scheduling/daos/UserDAO.java
#	src/main/java/com/frontleaves/scheduling/logic/UserLogic.java
#	src/main/java/com/frontleaves/scheduling/services/UserService.java

Commit:
refactor(user): 重构用户相关逻辑和数据访问对象

-将 checkUserExist 方法从 AuthLogic 移动到 UserLogic
- 优化 Redis 事务处理逻辑- 重构 UserDAO 中的 deleteUser 和 updateUser 方法
- 新增 UserLogic 中的 checkUserExist 实现
- 更新相关服务接口

xiaolfeng committed:
Merge branch 'develop' into feature/admin-manage-user

# Conflicts:
#	src/main/java/com/frontleaves/scheduling/daos/UserDAO.java

Commit:
refactor(user): 重构用户相关逻辑和数据访问对象

-将 checkUserExist 方法从 AuthLogic 移动到 UserLogic
- 优化 Redis 事务处理逻辑- 重构 UserDAO 中的 deleteUser 和 updateUser 方法
- 新增 UserLogic 中的 checkUserExist 实现
- 更新相关服务接口

xiaolfeng committed:
Merge branch 'develop' into feature/admin-manage-user

# Conflicts:
#	src/main/java/com/frontleaves/scheduling/controllers/UserController.java
#	src/main/java/com/frontleaves/scheduling/daos/StudentDAO.java
#	src/main/java/com/frontleaves/scheduling/daos/TeacherDAO.java
#	src/main/java/com/frontleaves/scheduling/logic/UserLogic.java
#	src/main/java/com/frontleaves/scheduling/services/UserService.java

xiaolfeng committed:
refactor(UserController): 优化 UserController 接口路径和参数

- 将 `getUserInfo` 的路径从 `/getUserInfo` 改为 `/user_uuid`，并使用 `@PathVariable` 替换 `@RequestParam`
- 将 `addUser` 的路径从 `/addUser` 改为 `/`
- 将 `deleteUser` 的路径从 `/deleteUser` 改为 `/user_uuid`，并使用 `@PathVariable` 替换 `@RequestParam`
- 将 `updateUser` 的路径从 `/updateUser` 改为 `/user_uuid`，并使用 `@PathVariable` 替换 `@RequestParam`
- 将 `getUserList` 的路径从 `/getUserList` 改为 `/list`
- 修正了部分方法的注释格式和代码格式

xiaolfeng commented on src/main/java/com/frontleaves/scheduling/logic/UserLogic.java:
```diff
@@ -79,4 +102,308 @@ public UserDO getUserByRequest(HttpServletRequest request) {
+            }
+            userDO.setRoleUuid(userEditVO.getRoleUuid());
+        }
+        if (!userEditVO.getPermission().isEmpty()) {
```
权限应该从数据库查询

xiaolfeng commented on src/main/java/com/frontleaves/scheduling/logic/UserLogic.java:
```diff
@@ -79,4 +102,308 @@ public UserDO getUserByRequest(HttpServletRequest request) {
+     * @return 用户数据对象
+     */
+    private UserDO exchangeOfUserData(UserEditVO userEditVO, UserDO userDO) {
+        if (!userEditVO.getName().isEmpty()) {
```
"status": false,
    "ban": true, 
问题

xiaolfeng committed:
feat(user): 添加UserAddInfoDTO并优化用户相关逻辑

- 新增 `UserAddInfoDTO` 类，用于处理用户新增信息，包括新密码。
- 优化 `UserDAO` 中的方法参数注释和方法调用方式。
- 更新 `UserController` 和 `UserService` 中的 `addUser` 方法返回类型为 `UserAddInfoDTO`。
- 在 `UserLogic` 中修改 `addUser` 方法，生成新的用户 UUID 并设置随机密码。
- 优化 `getUserList` 方法，增加对空结果的异常处理。
- 更新 `UserAddVO` 和 `StringConstant` 中的正则表达式，允许角色 UUID 为空。

xiaolfeng reviewed (changes requested):
错误修正后提交

Commit:
refactor(user): 重构用户权限相关代码

- 优化 PermissionDAO 中的权限查询逻辑
- 修改 PermissionDO 中的 desc 字段添加注解
- 重构 UserEditVO 中的权限表示方式
- 优化 UserLogic 中的权限检查和用户数据更新逻辑
